test/performance/user_test.rb

```diff
@@ -0,0 +1,42 @@
+require 'test_helper'
+
+require 'benchmark'
+
+module Performance
+  class UserTest < ActiveSupport::TestCase
+    class ConfigOffUser < User
+      self.allow_passwords_equal_to_email = true
+    end
+
+    class ConfigOnUser < User
+      self.allow_passwords_equal_to_email = false
+    end
+
+    test 'equal to email validation is performant for valid passwords' do
+      user1 = ConfigOffUser.new(email: 'speed@aol.com', password: 'Password1!')
+      user1.valid?
+      user2 = ConfigOnUser.new(email: 'speed@aol.com', password: 'Password1!')
+      user2.valid?
+
+      result1 = Benchmark.measure { user1.valid? }
+      result2 = Benchmark.measure { user2.valid? }
+
+      assert_in_delta(result1.real, result2.real, 0.005)
```

Comment:
Ok, I'm finding that this still fails every so often, so I'm thinking to either bump the delta way up or just get rid of this. Any opinions?

Reply:
I'm not sure how useful these tests are. You're basically checking to see if you do things twice that they happen about as fast, which is what I would expect for almost anything.

Reply:
This is because I want to make sure that the performance is the same regardless if the config is on or off. Before the change in this PR, the config being on resulted in this taking about 3x as long. 

Maybe this test is an over-reaction?

Reply:
Understood, and I have mixed feelings about it since that was the initial bug report.

Reply:
I can't think of a way to set this up without flakiness without a high enough delta that the test becomes meaningless. Unless you have any better ideas, I'll probably just remove it?

Reply:
I don't.  If there was enough need we could do a benchmark suite that would re-run the methods several thousand times to average out the noise, but I don't think that's the case normally.